src/grapheme.rs

```diff
@@ -362,19 +367,23 @@ impl<'a> DoubleEndedIterator for Graphemes<'a> {
         }
 
         self.catb = if take_curr {
-            None
+            // we need this anyway for Prepend lookahead,
+            // might as well cache it so that the next iteration can use it
+            self.string[..idx].chars().rev().next().map(gr::grapheme_category)
         } else  {
             idx = previdx;
             Some(cat)
         };
 
-        if self.extended && cat != gr::GC_Control {
-            // rule GB9b: include any preceding Prepend characters
-            for (i, c) in self.string[..idx].char_indices().rev() {
-                // TODO: Cache this to avoid repeated lookups in the common case.
-                match gr::grapheme_category(c) {
-                    gr::GC_Prepend => idx = i,
-                    _ => break
+        if let Some(gr::GC_Prepend) = self.catb {
+            if self.extended && cat != gr::GC_Control {
+                // rule GB9b: include any preceding Prepend characters
+                for (i, c) in self.string[..idx].char_indices().rev() {
+                    // TODO: Cache this to avoid repeated lookups in the common case.
+                    match gr::grapheme_category(c) {
+                        gr::GC_Prepend => idx = i,
```

Comment:
Hmm, this needs to either invalidate or update `self.catb`.

This also means we are missing a test case for this code, specifically for the case where multiple adjacent Prepend characters come between two "normal" characters.